firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionLifecycleNotifier.java

```diff
@@ -235,22 +235,27 @@ <A extends Activity> Task<Activity> getForegroundActivity(@Nullable Class<A> cla
         return Tasks.forResult(
             getActivityWithIgnoredClass(currentActivity, previousActivity, classToIgnore));
       }
+    }
 
-      TaskCompletionSource<Activity> task = new TaskCompletionSource<>();
+    TaskCompletionSource<Activity> task = new TaskCompletionSource<>();
 
-      addOnActivityResumedListener(
-          new OnActivityResumedListener() {
-            @Override
-            public void onResumed(Activity activity) {
-              synchronized (lock) {
-                task.setResult(
-                    getActivityWithIgnoredClass(activity, previousActivity, classToIgnore));
-              }
-              removeOnActivityResumedListener(this);
-            }
-          });
+    addOnActivityResumedListener(
+        new OnActivityResumedListener() {
+          @Override
+          public void onResumed(Activity activity) {
+            task.setResult(
+                getActivityWithIgnoredClass(activity, getPreviousActivity(), classToIgnore));
```

Comment:
This `setResult` call no longer happens inside a `synchronized (lock)`. Only the access on `previousActivity` does, below.

Reply:
But why? I don't see why the access to `previousActivity` needs to be protected by the `lock`. It's either `null` or not `null` - with or without the synchronization. So either `null` or not `null` gets passed as the second parameter to `getActivityWithIgnoredClass()` and that's fine, isn't it?

Reply:
You're right, but we need to wrap any access of `previousActivity` in a synchronized block because it's annotated by a `@GuardedBy`.